File: api/src/main/java/com/ruby/rubia_server/core/entity/WhatsAppInstance.java
package com.ruby.rubia_server.core.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.ruby.rubia_server.core.enums.MessagingProvider;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "whatsapp_instances")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(exclude = {"company"})
@EqualsAndHashCode(exclude = {"company"})
public class WhatsAppInstance {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    @JsonBackReference("company-whatsapp")
    private Company company;
    
    @Column(name = "phone_number", nullable = false)
    private String phoneNumber;
    
    @Column(name = "display_name")
    private String displayName;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "provider", nullable = false)
    @Builder.Default
    private MessagingProvider provider = MessagingProvider.Z_API;
    
    @Column(name = "instance_id")
    private String instanceId;
    
    @Column(name = "access_token")
    private String accessToken;
    
    @Column(name = "webhook_url")
    private String webhookUrl;
    
    @Column(name = "is_active")
    @Builder.Default
    private Boolean isActive = true;
    
    @Column(name = "is_primary")
    @Builder.Default
    private Boolean isPrimary = false;
    
    @Column(name = "configuration_data", columnDefinition = "TEXT")
    private String configurationData;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    // Helper methods
    public boolean isConfigured() {
        return instanceId != null && accessToken != null;
    }
    
    public boolean isConnected() {
        return isConfigured() && isActive;
    }
    
    public boolean needsConfiguration() {
        return !isConfigured();
    }
}

File: api-dotnet/Rubia.Server/Entities/WhatsAppInstance.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

[Table("whatsapp_instances")]
public class WhatsAppInstance : BaseEntity
{
    [Required]
    [Column("phone_number")]
    public string PhoneNumber { get; set; } = string.Empty;

    [Column("display_name")]
    public string? DisplayName { get; set; }

    [Column("provider")]
    [Required]
    public MessagingProvider Provider { get; set; } = MessagingProvider.ZApi;

    [Column("instance_id")]
    public string? InstanceId { get; set; }

    [Column("access_token")]
    public string? AccessToken { get; set; }

    [Column("webhook_url")]
    public string? WebhookUrl { get; set; }

    [Column("is_active")]
    public bool IsActive { get; set; } = true;

    [Column("is_primary")]
    public bool IsPrimary { get; set; } = false;

    [Column("configuration_data", TypeName = "TEXT")]
    public string? ConfigurationData { get; set; }

    // Navigation properties
    [Column("company_id")]
    [Required]
    public Guid CompanyId { get; set; }
    
    [ForeignKey("CompanyId")]
    public virtual Company Company { get; set; } = null!;

    // Helper methods
    public bool IsConfigured() => !string.IsNullOrEmpty(InstanceId) && !string.IsNullOrEmpty(AccessToken);
    
    public bool IsConnected() => IsConfigured() && IsActive;
    
    public bool NeedsConfiguration() => !IsConfigured();
}